### ISSUE TITLE
valgrind: kvs leaks fence_t objects

### ISSUE THREAD

**Issue by grondo** [flux-framework/flux-core#826]
From valgrind, it appears kvs leaks memory `calloc`'d for fence objects. Not submitting a PR in case I got the problem or the fix incorrect

```diff
diff --git a/src/modules/kvs/kvs.c b/src/modules/kvs/kvs.c
index a872c3b..87a1c44 100644
--- a/src/modules/kvs/kvs.c
+++ b/src/modules/kvs/kvs.c
@@ -1070,6 +1070,7 @@ static void fence_destroy (fence_t *f)
             /* FIXME: respond with error here? */
             zlist_destroy (&f->requests);
         }
+        free (f);
     }
 }
```

**Comment by garlick** [comment]
Thanks, good catch!  I'm tacking this fix onto pr #824

**Comment by grondo** [comment]
Great thanks!
